src/test/java/org/takes/tk/TkRedirectTest.java

```diff
@@ -94,4 +95,30 @@ public void createsRedirectResponseWithUrlAndStatus() throws IOException {
         );
     }
 
+    /**
+     * TkRedirect should carry on the query and the fragment.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void redirectCarriesQueryAndFragment() throws IOException {
+        MatcherAssert.assertThat(
+            new RsPrint(
+                new TkRedirect("http://www.google.com/abc?b=2").act(
+                    new RqFake("GET", "/hi?a=1#test")
+                )
+            ).print(),
+            Matchers.equalTo(
```

Comment:
@neonailol it would be clearer to use `HmRsHeader("Location", ...)` here

Reply:
@g4s8 Updated the test